drag/drag.view.ts

```diff
@@ -1,41 +1,181 @@
 namespace $.$$ {
-	
-	/**
-	 * @see https://mol.hyoo.ru/#!section=demos/demo=mol_drag_demo
-	 */
-	export class $mol_drag extends $.$mol_drag {
-
-		@ $mol_mem
-		status( next = 'ready' as 'ready' | 'drag' ) { return next }
-
-		drag_start( event : DragEvent ) {
-
-			setTimeout( ()=> this.status( 'drag' ) )
-
-			const transfer = this.transfer()
-			for( let type in transfer ) {
-				event.dataTransfer!.setData( type , transfer[ type as keyof typeof transfer] )
-			}
-
-			event.dataTransfer!.setDragImage( this.image() , 0 , -32 )
-			
-			const effects = [] as string[]
-			if( this.allow_copy() ) effects.push( 'Copy' )
-			if( this.allow_link() ) effects.push( 'Link' )
-			if( this.allow_move() ) effects.push( 'Move' )
-
-			let effectAllowed = effects[0].toLowerCase() + effects.slice(1).join('')
-			if( effectAllowed === 'copyLinkMove' ) effectAllowed = 'all'
-			event.dataTransfer!.effectAllowed = effectAllowed as DataTransfer['effectAllowed']
-			
-			this.start( event )
-			
-		}
-		
-		drag_end( event : DragEvent ) {
-			setTimeout( ()=> this.status( 'ready' ) )
-			this.end( event )
-		}
-
-	}
-}
+    /**
+     * @see https://mol.hyoo.ru/#!section=demos/demo=mol_drag_demo
+     */
+    export class $mol_drag extends $.$mol_drag {
+      private dragged_task: HTMLElement | null = null;
+      private initial_task_index: number | null = null;
+      private draggedOverElem: HTMLElement | null = null;
```

Comment:
There should be no info about the dragged element, this is the work for `$mol_drop`. The new position of the passed element must be determined by `$mol_drop`. `$mol_drag` don't move element before drop. 

If we want a component that immediately changes the position of elements when the pointer is moved, it will be a different component, not `$mol_drag`.